lessons/projects/5.mdx

```diff
@@ -0,0 +1,1379 @@
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-evenly',
+  gap: '50px',
+}
+
+const NFTCard = {
+  display: 'flex',
+  flexDirection: 'column',
+  border: '2px solid white',
+  borderRadius: '10px',
+  padding: '20px',
+  alignItems: 'center',
+  gap: '10px',
+  fontWeight: 'bold',
+}
+
+const NFTMint = {
+  fontWeight: '700',
+  padding: '5px 20px',
+  border: '2px solid white',
+  color: 'white',
+  backgroundColor: 'black',
+  borderRadius: '5px',
+  cursor: 'pointer',
+}
+
+const modal = {
+  position: 'fixed',
+  left: '0',
+  top: '0',
+  right: '0',
+  bottom: '0',
+  backgroundColor: 'rgba(0, 0, 0, 0.8)',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  textAlign: 'center',
+  zIndex: '1',
+}
+
+const modalContent = {
+  backgroundColor: '#fff',
+  padding: '10px 30px',
+  borderRadius: '16px',
+  color: '#000'
+}
+
+const modalBody = {
+  padding: '20px',
+  borderTop: '1px solid #eee',
+  borderBottom: '1px solid #eee',
+}
+
+const modalFooter = {
+  display: 'flex',
+  flexDirection: 'column',
+  gap: '10px',
+  padding: '20px',
+  justifyContent: 'space-evenly'
+}
+
+const modalButton = {
+  padding: '10px 20px',
+  backgroundColor: '#fff',
+  color: '#666',
+  border: '0',
+  borderRadius: '10px',
+  fontSize: '18px',
+  fontWeight: '700',
+  boxShadow: '0 0.2em 0.4em 0 rgba(0, 0, 0, 0.2), 0 0.3em 1em 0 rgba(0, 0, 0, 0.19)',
+}
+```
+</ContentSideDrawer>
+<br/>
+
+Take a breather and stretch yourself before we do the last bit of coding.
+
+### Successful Mint Modal
+
+Doesn't it feel like we are still missing something?
+
+Right! We want our users to see information when they minting. We want 
+them to know when a mint *is in progress* and we want them to see the 
+result once the mint *is successful*.
+
+But what is this *modal*. It is a an element that is usually used as 
+an overlay that provides its own interaction environment separate from 
+the page below it.
+
+Okay lets write some code for it.
+
+To be able to display information about our latest mint, we must first
+get information about it from our smart contract.
+
+```jsx
+// imports, consts and WAGMI useContractWrite hook...
+
+  const {
+    data: tokenData,
+    refetch: refetchTokenData,
+  } = useContractRead({
+    addressOrName: CONTRACT_ADDRESS,
+    contractInterface: TierABI.abi,
+    functionName: "totalSupply",
+    watch: true,
+  })
+
+  const {
+    data: tokenURI,
+  } = useContractRead({
+    addressOrName: CONTRACT_ADDRESS,
+    contractInterface: TierABI.abi,
+    functionName: "tokenURI",
+    args: tokenData,
+    watch: true,
+  })
+
+  // useEffect to check user connection...
+
+  useEffect(() => {
+    try {
+      if (tokenURI) {
+        setLatestNFTMinted(JSON.parse(window.atob(tokenURI.substring(tokenURI.indexOf(',') + 1))));
+      }
+    } catch (error) {
+      console.log("Error fetching token URI", error.message);
+    }
+  }, [tokenData, tokenURI]);
+
+  const mintToken = async (e) => {
```

Comment:
it starts to get extra messy here trying to follow the commented out directions.